spec/zint/barcode_spec.rb

```diff
@@ -132,13 +132,51 @@ module Zint
       it "exports barcode as vector from given value" do
         vector_struct = barcode.to_vector
 
-        expect(vector_struct.is_a?(Zint::Structs::Vector)).to be true
+        expect(vector_struct.is_a?(Structs::Vector)).to be true
       end
 
       it "exports barcode as vector from input file" do
         vector_struct = barcode_with_input_file.to_vector
 
-        expect(vector_struct.is_a?(Zint::Structs::Vector)).to be true
+        expect(vector_struct.is_a?(Structs::Vector)).to be true
+      end
+
+      it "exports barcode as vector of strings" do
+        v = described_class.new(value: "ABC").to_vector
+        strs = v.each_string.to_a
+        expect(strs.size).to eq(1)
+        expect(strs[0].text).to eq("ABC")
+        expect(strs[0].x).to be > 10.0
+        expect(strs[0].y).to be > 10.0
+        expect(strs[0].fsize).to eq(14)
+        expect(strs[0].width).to be > 10.0
+        expect(strs[0].halign).to eq(0)
+        expect(strs[0].rotation).to eq(0)
+      end
+
+      it "exports barcode as vector of rectangles" do
+        v = described_class.new(value: "ABC").to_vector
+        strs = v.each_rectangle.to_a
+        expect(strs.size).to eq(19)
+        expect(strs[0].x).to be >= 0.0
+        expect(strs[0].y).to be >= 0.0
+        expect(strs[0].height).to be > 1.0
+        expect(strs[0].width).to be > 1.0
+        expect(strs[0].colour).to eq(-1)
+      end
+
+      it "exports barcode as vector of circles" do
+        v = described_class.new(value: "ABC").to_vector
+        strs = v.each_circle.to_a
+        expect(strs.size).to eq(0)
+        # TODO use a barcode with circles
```

Comment:
@api-walker Which symbol type could be used for testing circles and hexagons?

Reply:
Hi Lars,
please have a look at BARCODE_MAXICODE.

It looks like this:
![maxicode_2 1](https://user-images.githubusercontent.com/11318229/235650633-177978fe-df67-499f-b2b6-63de0ec78ee1.png)

Reply:
OK, thanks! I added specs for hexagons and circles now.